New test server

andrew-aladev opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/thaliproject/thali_cordovaplugin/1007)
<!-- Reviewable:end -->

ThaliTester commented:
PR is added to the queue for testing as 4. task. (8046b77)

msftclas commented:
Hi __@andrew-aladev__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

ThaliTester commented:
Test (Fail) 84500654 build is completed (8046b77)

See https://github.com/ThaliTester/TestResults/tree/845006548046b77_New_test_server_andrew-aladev/ for the logs

artemjackson commented:
@andrew-aladev Assign this please to me so I can review it

ThaliTester commented:
PR is added to the queue for testing as 1. task. (833e1a9)

ThaliTester commented:
Test (Fail) 84500654 build is completed (833e1a9)

See https://github.com/ThaliTester/TestResults/tree/84500654833e1a9_New_test_server_andrew-aladev/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (1145c7f)

ThaliTester commented:
Test (Success) 84500654 build is completed (1145c7f)

See https://github.com/ThaliTester/TestResults/tree/845006541145c7f_New_test_server_andrew-aladev/ for the logs

ThaliTester commented:
Test 845006541145c7f(1145c7f) has failed

See https://github.com/ThaliTester/TestResults/tree/845006541145c7f_New_test_server_andrew-aladev/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (8dec34d)

andrew-aladev commented:
Here we can update `socket.io` to the latest version. Closes #1008

ThaliTester commented:
Build is completed (Fail) but couldn't commit the log file. (8dec34d)

commit_logs.sh: push origin has failed : 
 [0m

warning: push.default is unset; its implicit value has changed in
Git 2.0 from 'matching' to 'simple'. To squelch this message
and maintain the traditional behavior, use:

  git config --global push.default matching

To squelch this message and adopt the new behavior now, use:

  git config --global push.default simple

When push.default is set to 'matching', git will push local branches
to the remote branches that already exist with the same name.

Since Git 2.0, Git defaults to the more conservative 'simple'
behavior, which only pushes the current branch to the corresponding
remote branch that 'git pull' uses to update the current branch.

See 'git help config' and search for 'push.default' for further information.
(the 'simple' mode was introduced in Git 1.7.11. Use the similar mode
'current' instead of 'simple' if you sometimes use older versions of Git)

fatal: unable to access 'https://github.com/ThaliTester/TestResults/': Failed to connect to github.com port 443: Operation timed out
Switched to branch 'master'
Switched to a new branch '845006548dec34d_New_test_server_andrew-aladev'
fatal: unable to access 'https://github.com/ThaliTester/TestResults/': Failed to connect to github.com port 443: Operation timed out
rm: test: No such file or directory


Error: Command failed: warning: push.default is unset; its implicit value has changed in
Git 2.0 from 'matching' to 'simple'. To squelch this message
and maintain the traditional behavior, use:

  git config --global push.default matching

To squelch this message and adopt the new behavior now, use:

  git config --global push.default simple

When push.default is set to 'matching', git will push local branches
to the remote branches that already exist with the same name.

Since Git 2.0, Git defaults to the more conservative 'simple'
behavior, which only pushes the current branch to the corresponding
remote branch that 'git pull' uses to update the current branch.

See 'git help config' and search for 'push.default' for further information.
(the 'simple' mode was introduced in Git 1.7.11. Use the similar mode
'current' instead of 'simple' if you sometimes use older versions of Git)

fatal: unable to access 'https://github.com/ThaliTester/TestResults/': Failed to connect to github.com port 443: Operation timed out
Switched to branch 'master'
Switched to a new branch '845006548dec34d_New_test_server_andrew-aladev'
fatal: unable to access 'https://github.com/ThaliTester/TestResults/': Failed to connect to github.com port 443: Operation timed out
rm: test: No such file or directory


Error: Command failed: npm http request GET http://192.168.1.100:4873/fs-extra-promise
npm http request GET http://192.168.1.100:4873/socket.io
npm http request GET http://192.168.1.100:4873/express
npm http request GET http://192.168.1.100:4873/bluebird
npm http request GET http://192.168.1.100:4873/winston
npm http 200 http://192.168.1.100:4873/fs-extra-promise
npm http 200 http://192.168.1.100:4873/winston
npm http 200 http://192.168.1.100:4873/socket.io
npm http 200 http://192.168.1.100:4873/express
npm http 200 http://192.168.1.100:4873/bluebird
npm http fetch GET http://192.168.1.100:4873/bluebird/-/bluebird-3.4.6.tgz
npm http fetch 200 http://192.168.1.100:4873/bluebird/-/bluebird-3.4.6.tgz
npm http request GET http://192.168.1.100:4873/accepts
npm http request GET http://192.168.1.100:4873/array-flatten
npm http request GET http://192.168.1.100:4873/content-type
npm http request GET http://192.168.1.100:4873/content-disposition
npm http request GET http://192.168.1.100:4873/cookie
npm http request GET http://192.168.1.100:4873/cookie-signature
npm http request GET http://192.168.1.100:4873/depd
npm http request GET http://192.168.1.100:4873/debug
npm http request GET http://192.168.1.100:4873/escape-html
npm http request GET http://192.168.1.100:4873/etag
npm http request GET http://192.168.1.100:4873/finalhandler
npm http request GET http://192.168.1.100:4873/merge-descriptors
npm http request GET http://192.168.1.100:4873/fresh
npm http request GET http://192.168.1.100:4873/methods
npm http request GET http://192.168.1.100:4873/parseurl
npm http request GET http://192.168.1.100:4873/on-finished
npm http request GET http://192.168.1.100:4873/proxy-addr
npm http request GET http://192.168.1.100:4873/path-to-regexp
npm http request GET http://192.168.1.100:4873/range-parser
npm http request GET http://192.168.1.100:4873/qs
npm http request GET http://192.168.1.100:4873/serve-static
npm http request GET http://192.168.1.100:4873/send
npm http request GET http://192.168.1.100:4873/utils-merge
npm http request GET http://192.168.1.100:4873/type-is
npm http request GET http://192.168.1.100:4873/vary
npm http 200 http://192.168.1.100:4873/array-flatten
npm http 200 http://192.168.1.100:4873/accepts
npm http 200 http://192.168.1.100:4873/content-disposition
npm http 200 http://192.168.1.100:4873/content-type
npm http 200 http://192.168.1.100:4873/cookie
npm http 200 http://192.168.1.100:4873/depd
npm http 200 http://192.168.1.100:4873/cookie-signature
npm http 200 http://192.168.1.100:4873/debug
npm http 200 http://192.168.1.100:4873/etag
npm http 200 http://192.168.1.100:4873/escape-html
npm http 200 http://192.168.1.100:4873/finalhandler
npm http 200 http://192.168.1.100:4873/merge-descriptors
npm http 200 http://192.168.1.100:4873/parseurl
npm http 200 http://192.168.1.100:4873/fresh
npm http 200 http://192.168.1.100:4873/methods
npm http 200 http://192.168.1.100:4873/on-finished
npm http 200 http://192.168.1.100:4873/proxy-addr
npm http 200 http://192.168.1.100:4873/range-parser
npm http 200 http://192.168.1.100:4873/qs
npm http 200 http://192.168.1.100:4873/path-to-regexp
npm http 200 http://192.168.1.100:4873/serve-static
npm http 200 http://192.168.1.100:4873/send
npm http 200 http://192.168.1.100:4873/utils-merge
npm http 200 http://192.168.1.100:4873/type-is
npm http 200 http://192.168.1.100:4873/vary
npm http request GET http://192.168.1.100:4873/negotiator
npm http request GET http://192.168.1.100:4873/mime-types
npm http 200 http://192.168.1.100:4873/negotiator
npm http 200 http://192.168.1.100:4873/mime-types
npm http request GET http://192.168.1.100:4873/mime-db
npm http 200 http://192.168.1.100:4873/mime-db
npm http request GET http://192.168.1.100:4873/ms
npm http 200 http://192.168.1.100:4873/ms
npm http request GET http://192.168.1.100:4873/unpipe
npm http 200 http://192.168.1.100:4873/unpipe
npm http request GET http://192.168.1.100:4873/ee-first
npm http 200 http://192.168.1.100:4873/ee-first
npm http request GET http://192.168.1.100:4873/forwarded
npm http request GET http://192.168.1.100:4873/ipaddr.js
npm http 200 http://192.168.1.100:4873/ipaddr.js
npm http 200 http://192.168.1.100:4873/forwarded
npm http request GET http://192.168.1.100:4873/destroy
npm http request GET http://192.168.1.100:4873/statuses
npm http request GET http://192.168.1.100:4873/http-errors
npm http request GET http://192.168.1.100:4873/mime
npm http 200 http://192.168.1.100:4873/mime
npm http 200 http://192.168.1.100:4873/destroy
npm http 200 http://192.168.1.100:4873/statuses
npm http 200 http://192.168.1.100:4873/http-errors
npm http request GET http://192.168.1.100:4873/inherits
npm http 200 http://192.168.1.100:4873/inherits
npm http fetch GET http://192.168.1.100:4873/inherits/-/inherits-2.0.3.tgz
npm http fetch 200 http://192.168.1.100:4873/inherits/-/inherits-2.0.3.tgz
npm http request GET http://192.168.1.100:4873/media-typer
npm http 200 http://192.168.1.100:4873/media-typer
npm http request GET http://192.168.1.100:4873/fs-extra
npm http fetch GET http://192.168.1.100:4873/bluebird/-/bluebird-2.11.0.tgz
npm http fetch 200 http://192.168.1.100:4873/bluebird/-/bluebird-2.11.0.tgz
npm http 200 http://192.168.1.100:4873/fs-extra
npm http request GET http://192.168.1.100:4873/graceful-fs
npm http request GET http://192.168.1.100:4873/jsonfile
npm http request GET http://192.168.1.100:4873/rimraf
npm http 200 http://192.168.1.100:4873/graceful-fs
npm http 200 http://192.168.1.100:4873/jsonfile
npm http fetch GET http://192.168.1.100:4873/graceful-fs/-/graceful-fs-4.1.6.tgz
npm http fetch 200 http://192.168.1.100:4873/graceful-fs/-/graceful-fs-4.1.6.tgz
npm http 200 http://192.168.1.100:4873/rimraf
npm http request GET http://192.168.1.100:4873/glob
npm http 200 http://192.168.1.100:4873/glob
npm http fetch GET http://192.168.1.100:4873/glob/-/glob-7.0.6.tgz
npm http fetch 200 http://192.168.1.100:4873/glob/-/glob-7.0.6.tgz
npm http request GET http://192.168.1.100:4873/fs.realpath
npm http request GET http://192.168.1.100:4873/inflight
npm http request GET http://192.168.1.100:4873/path-is-absolute
npm http request GET http://192.168.1.100:4873/once
npm http request GET http://192.168.1.100:4873/minimatch
npm http 200 http://192.168.1.100:4873/fs.realpath
npm http 200 http://192.168.1.100:4873/inflight
npm http 200 http://192.168.1.100:4873/path-is-absolute
npm http 200 http://192.168.1.100:4873/once
npm http 200 http://192.168.1.100:4873/minimatch
npm http fetch GET http://192.168.1.100:4873/once/-/once-1.4.0.tgz
npm http fetch 200 http://192.168.1.100:4873/once/-/once-1.4.0.tgz
npm http request GET http://192.168.1.100:4873/wrappy
npm http 200 http://192.168.1.100:4873/wrappy
npm http request GET http://192.168.1.100:4873/brace-expansion
npm http 200 http://192.168.1.100:4873/brace-expansion
npm http request GET http://192.168.1.100:4873/balanced-match
npm http request GET http://192.168.1.100:4873/concat-map
npm http 200 http://192.168.1.100:4873/balanced-match
npm http 200 http://192.168.1.100:4873/concat-map
npm http request GET http://192.168.1.100:4873/engine.io
npm http request GET http://192.168.1.100:4873/has-binary
npm http request GET http://192.168.1.100:4873/socket.io-adapter
npm http request GET http://192.168.1.100:4873/socket.io-parser
npm http request GET http://192.168.1.100:4873/socket.io-client
npm http 200 http://192.168.1.100:4873/has-binary
npm http 200 http://192.168.1.100:4873/socket.io-adapter
npm http 200 http://192.168.1.100:4873/socket.io-parser
npm http 200 http://192.168.1.100:4873/engine.io
npm http 200 http://192.168.1.100:4873/socket.io-client
npm http request GET http://192.168.1.100:4873/base64id
npm http request GET http://192.168.1.100:4873/ws
npm http request GET http://192.168.1.100:4873/engine.io-parser
npm http 200 http://192.168.1.100:4873/base64id
npm http 200 http://192.168.1.100:4873/engine.io-parser
npm http 200 http://192.168.1.100:4873/ws
npm http request GET http://192.168.1.100:4873/arraybuffer.slice
npm http request GET http://192.168.1.100:4873/base64-arraybuffer
npm http request GET http://192.168.1.100:4873/after
npm http request GET http://192.168.1.100:4873/blob
npm http request GET http://192.168.1.100:4873/utf8
npm http 200 http://192.168.1.100:4873/arraybuffer.slice
npm http 200 http://192.168.1.100:4873/utf8
npm http 200 http://192.168.1.100:4873/blob
npm http 200 http://192.168.1.100:4873/base64-arraybuffer
npm http 200 http://192.168.1.100:4873/after
npm http request GET http://192.168.1.100:4873/isarray
npm http 200 http://192.168.1.100:4873/isarray
npm http request GET http://192.168.1.100:4873/options
npm http request GET http://192.168.1.100:4873/ultron
npm http 200 http://192.168.1.100:4873/options
npm http 200 http://192.168.1.100:4873/ultron
npm http request GET http://192.168.1.100:4873/json3
npm http request GET http://192.168.1.100:4873/component-emitter
npm http request GET http://192.168.1.100:4873/benchmark
npm http 200 http://192.168.1.100:4873/component-emitter
npm http 200 http://192.168.1.100:4873/benchmark
npm http 200 http://192.168.1.100:4873/json3
npm http request GET http://192.168.1.100:4873/object-component
npm http request GET http://192.168.1.100:4873/component-bind
npm http request GET http://192.168.1.100:4873/engine.io-client
npm http request GET http://192.168.1.100:4873/indexof
npm http request GET http://192.168.1.100:4873/parseuri
npm http request GET http://192.168.1.100:4873/backo2
npm http request GET http://192.168.1.100:4873/to-array
npm http 200 http://192.168.1.100:4873/indexof
npm http 200 http://192.168.1.100:4873/component-bind
npm http 200 http://192.168.1.100:4873/object-component
npm http 200 http://192.168.1.100:4873/parseuri
npm http 200 http://192.168.1.100:4873/engine.io-client
npm http 200 http://192.168.1.100:4873/to-array
npm http 200 http://192.168.1.100:4873/backo2
npm http request GET http://192.168.1.100:4873/has-cors
npm http request GET http://192.168.1.100:4873/xmlhttprequest-ssl
npm http request GET http://192.168.1.100:4873/parsejson
npm http request GET http://192.168.1.100:4873/parseqs
npm http request GET http://192.168.1.100:4873/yeast
npm http request GET http://192.168.1.100:4873/component-inherit
npm http 200 http://192.168.1.100:4873/has-cors
npm http 200 http://192.168.1.100:4873/parseqs
npm http 200 http://192.168.1.100:4873/yeast
npm http 200 http://192.168.1.100:4873/xmlhttprequest-ssl
npm http 200 http://192.168.1.100:4873/parsejson
npm http 200 http://192.168.1.100:4873/component-inherit
npm http request GET http://192.168.1.100:4873/better-assert
npm http 200 http://192.168.1.100:4873/better-assert
npm http request GET http://192.168.1.100:4873/callsite
npm http 200 http://192.168.1.100:4873/callsite
npm http request GET http://192.168.1.100:4873/async
npm http request GET http://192.168.1.100:4873/eyes
npm http request GET http://192.168.1.100:4873/cycle
npm http request GET http://192.168.1.100:4873/colors
npm http request GET http://192.168.1.100:4873/isstream
npm http request GET http://192.168.1.100:4873/pkginfo
npm http request GET http://192.168.1.100:4873/stack-trace
npm http 200 http://192.168.1.100:4873/cycle
npm http 200 http://192.168.1.100:4873/eyes
npm http 200 http://192.168.1.100:4873/isstream
npm http 200 http://192.168.1.100:4873/colors
npm http 200 http://192.168.1.100:4873/async
npm http 200 http://192.168.1.100:4873/pkginfo
npm http 200 http://192.168.1.100:4873/stack-trace
npm http request GET http://192.168.1.100:4873/node-uuid
npm http request GET http://192.168.1.100:4873/tape
npm http 200 http://192.168.1.100:4873/node-uuid
npm http 200 http://192.168.1.100:4873/tape
npm http request GET http://192.168.1.100:4873/jsonify
npm http request GET http://192.168.1.100:4873/defined
npm http request GET http://192.168.1.100:4873/deep-equal
npm http request GET http://192.168.1.100:4873/through
npm http 200 http://192.168.1.100:4873/deep-equal
npm http 200 http://192.168.1.100:4873/jsonify
npm http 200 http://192.168.1.100:4873/defined
npm http 200 http://192.168.1.100:4873/through
npm WARN EPACKAGEJSON thali-test-server@0.0.1 No repository field.
npm WARN EPACKAGEJSON thali-test-server@0.0.1 No license field.
npm http request GET http://192.168.1.100:4873/express
npm http request GET http://192.168.1.100:4873/body-parser
npm http request GET http://192.168.1.100:4873/forever-agent
npm http request GET http://192.168.1.100:4873/ip
npm http request GET http://192.168.1.100:4873/winston
npm http request GET http://192.168.1.100:4873/javascript-state-machine
npm http request GET http://192.168.1.100:4873/long
npm http request GET http://192.168.1.100:4873/lie
npm http request GET http://192.168.1.100:4873/multiplex
npm http request GET http://192.168.1.100:4873/request
npm http request GET http://192.168.1.100:4873/salti
npm http request GET http://192.168.1.100:4873/urlsafe-base64
npm http 304 http://192.168.1.100:4873/express
npm http 304 http://192.168.1.100:4873/winston
npm http 200 http://192.168.1.100:4873/forever-agent
npm http 200 http://192.168.1.100:4873/ip
npm http 200 http://192.168.1.100:4873/body-parser
npm http 200 http://192.168.1.100:4873/long
npm http 200 http://192.168.1.100:4873/request
npm http 200 http://192.168.1.100:4873/lie
npm http 200 http://192.168.1.100:4873/urlsafe-base64
npm http 304 http://192.168.1.100:4873/salti
npm http 200 http://192.168.1.100:4873/javascript-state-machine
npm http 200 http://192.168.1.100:4873/multiplex
npm ERR! git fetch -a origin (git://github.com/thaliproject/node-ssdp.git) fatal: Unable to look up github.com (port 9418) (nodename nor servname provided, or not known)
npm ERR! git clone --template=/Users/thali/.npm/_git-remotes/_templates --mirror git@github.com:thaliproject/node-ssdp.git /Users/thali/.npm/_git-remotes/git-github-com-thaliproject-node-ssdp-git-fcc5d403f6cc5e667572fba059332908c3cc822e-719e5541de2e849a62857f2fe61a9cdb: Cloning into bare repository '/Users/thali/.npm/_git-remotes/git-github-com-thaliproject-node-ssdp-git-fcc5d403f6cc5e667572fba059332908c3cc822e-719e5541de2e849a62857f2fe61a9cdb'...
npm ERR! git clone --template=/Users/thali/.npm/_git-remotes/_templates --mirror git@github.com:thaliproject/node-ssdp.git /Users/thali/.npm/_git-remotes/git-github-com-thaliproject-node-ssdp-git-fcc5d403f6cc5e667572fba059332908c3cc822e-719e5541de2e849a62857f2fe61a9cdb: ssh: Could not resolve hostname github.com: nodename nor servname provided, or not known
npm ERR! git clone --template=/Users/thali/.npm/_git-remotes/_templates --mirror git@github.com:thaliproject/node-ssdp.git /Users/thali/.npm/_git-remotes/git-github-com-thaliproject-node-ssdp-git-fcc5d403f6cc5e667572fba059332908c3cc822e-719e5541de2e849a62857f2fe61a9cdb: fatal: Could not read from remote repository.
npm ERR! git clone --template=/Users/thali/.npm/_git-remotes/_templates --mirror git@github.com:thaliproject/node-ssdp.git /Users/thali/.npm/_git-remotes/git-github-com-thaliproject-node-ssdp-git-fcc5d403f6cc5e667572fba059332908c3cc822e-719e5541de2e849a62857f2fe61a9cdb: 
npm ERR! git clone --template=/Users/thali/.npm/_git-remotes/_templates --mirror git@github.com:thaliproject/node-ssdp.git /Users/thali/.npm/_git-remotes/git-github-com-thaliproject-node-ssdp-git-fcc5d403f6cc5e667572fba059332908c3cc822e-719e5541de2e849a62857f2fe61a9cdb: Please make sure you have the correct access rights
npm ERR! git clone --template=/Users/thali/.npm/_git-remotes/_templates --mirror git@github.com:thaliproject/node-ssdp.git /Users/thali/.npm/_git-remotes/git-github-com-thaliproject-node-ssdp-git-fcc5d403f6cc5e667572fba059332908c3cc822e-719e5541de2e849a62857f2fe61a9cdb: and the repository exists.
npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/bin/jx" "/Users/thali/.jx/npm/bin/npm-cli.js" "--loglevel" "http" "--color" "true" "install" "--no-optional" "--build-from-source"
npm ERR! node v0.10.40
npm ERR! npm  v3.3.12
npm ERR! code 128

npm ERR! Command failed: Cloning into bare repository '/Users/thali/.npm/_git-remotes/git-github-com-thaliproject-node-ssdp-git-fcc5d403f6cc5e667572fba059332908c3cc822e-719e5541de2e849a62857f2fe61a9cdb'...
npm ERR! ssh: Could not resolve hostname github.com: nodename nor servname provided, or not known
npm ERR! fatal: Could not read from remote repository.
npm ERR! 
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm ERR! 
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/thali/Github/Thali_CordovaPlugin/thali/npm-debug.log

ThaliTester commented:
PR is added to the queue for testing as 4. task. (1b10566)

ThaliTester commented:
PR is added to the queue for testing as 6. task. (1a293f3)

ThaliTester commented:
Test (Success) 84500654 build is completed (1b10566)

See https://github.com/ThaliTester/TestResults/tree/845006541b10566_New_test_server_andrew-aladev/ for the logs

andrew-aladev commented:
Done

artemjackson commented on test/TestServer/TestDevice.js:
```diff
@@ -1,19 +1,255 @@
+
+  this.btAddress = data.btaddress;
+
+  this._options = extend({}, TestDevice.defaults);
```
extend can process multiple parameters like that: `extend({}, a,b,c)`

ThaliTester commented:
Test (Success) 84500654 build is completed (1a293f3)

See https://github.com/ThaliTester/TestResults/tree/845006541a293f3_New_test_server_andrew-aladev/ for the logs

ThaliTester commented:
Test 845006541b10566(1b10566) has failed

See https://github.com/ThaliTester/TestResults/tree/845006541b10566_New_test_server_andrew-aladev/ for the fail logs

ThaliTester commented:
Test 845006541a293f3(1a293f3) has failed

See https://github.com/ThaliTester/TestResults/tree/845006541a293f3_New_test_server_andrew-aladev/ for the fail logs

artemjackson commented:
Reviewed 1 of 13 files at r1, 1 of 1 files at r3, 1 of 2 files at r4, 1 of 5 files at r5, 1 of 2 files at r6.
Review status: 5 of 18 files reviewed at latest revision, 3 unresolved discussions.

---

*[test/TestServer/index.js, line 73 at r6](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRXqWEhFKpbkZczdwBj:-KRXqWEhFKpbkZczdwBk:b-wr886b) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1a293f3a78e4099d23156379bee6f935d1517e8f/test/TestServer/index.js#L73)):*
> ```JavaScript
> });
> 
> var RETRY_COUNT = 120;
> ```

Suggest you move it to config file

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007)*
<!-- Sent from Reviewable.io -->

andrew-aladev commented:
I moved this stuff to config file. I had a problem about `extend`:

```
jx
> require('util')._extend({}, { a : 1 }, { a : 2 })
{ a: 1 }
> require('object-assign')({}, { a : 1 }, { a : 2 })
{ a: 2 }
```

So I changed `extend` to `objectAssign`.

ThaliTester commented:
PR is added to the queue for testing as 3. task. (64529c3)

ThaliTester commented:
Test (Success) 84500654 build is completed (64529c3)

See https://github.com/ThaliTester/TestResults/tree/8450065464529c3_New_test_server_andrew-aladev/ for the logs

ThaliTester commented:
Test 8450065464529c3(64529c3) has failed

See https://github.com/ThaliTester/TestResults/tree/8450065464529c3_New_test_server_andrew-aladev/ for the fail logs

artemjackson commented:
Reviewing of this PR stolen a lot of my time particularly because of big files.
So my suggestion is to split the files where it's possible into smaller ones. It means to split huge classes(like `TestServer/index.js` or `TestDevice`) into smaller components for better maintainability 

---

Reviewed 4 of 13 files at r1, 7 of 7 files at r2, 9 of 9 files at r7.
Review status: all files reviewed at latest revision, 20 unresolved discussions.

---

*[test/TestServer/index.js, line 12 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRhqD-nD2Xii7FKk_cZ:-KRhqD-nD2Xii7FKk_c_:b-mrdk1x) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/64529c3e7c7f6b32f9a806dd80d3d74bebe965de/test/TestServer/index.js#L12)):*
> ```JavaScript
> var socketIO = require('socket.io');
> var winston  = require('winston');
> var Promise  = require('bluebird');
> ```

It would be better to some abstraction like `ThaliPromise` for wrapping actual `Promise` library and it's configuration.
E.g. please see #973.

---

*[test/TestServer/index.js, line 28 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRhuFzc-55WN2qHBYyz:-KRhuFzc-55WN2qHBYz-:b6dwa3w) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/64529c3e7c7f6b32f9a806dd80d3d74bebe965de/test/TestServer/index.js#L28)):*
> ```JavaScript
> });
> 
> var logger = new winston.Logger({
> ```

It would be better to move logger initialization and configuration in separate file and just import it where needed

---

*[test/TestServer/index.js, line 42 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRhuR72fP5ITnDYR38w:-KRhuR72fP5ITnDYR38x:bag5uuz) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/64529c3e7c7f6b32f9a806dd80d3d74bebe965de/test/TestServer/index.js#L42)):*
> ```JavaScript
> });
> 
> var server = http.createServer();
> ```

I would suggest to move server initialization and configuration to separate file and provide an interface for dealing with it

---

*[test/TestServer/index.js, line 50 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRhusaZacUEna7Pmgip:-KRhusa_DtDZiU2q_GEO:b2kbfd0) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/64529c3e7c7f6b32f9a806dd80d3d74bebe965de/test/TestServer/index.js#L50)):*
> ```JavaScript
> });
> 
> process
> ```

Once server and logger are in separated files it will be possible to move these handlers to some file to not pollute this one

---

*[test/TestServer/index.js, line 113 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRhwas-9FCLBwkc5lOW:-KRhwas-9FCLBwkc5lOX:b-jvp5mw) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/64529c3e7c7f6b32f9a806dd80d3d74bebe965de/test/TestServer/index.js#L113)):*
> ```JavaScript
>         error.toString(), error.stack
>       );
>       io.close();
> ```

I noticed that there are a lot of usages of
```
io.close();
process.exit(code);
```
Probably this should be moved to `function`

BTW if server is in separated module it could be `function` of that module

---

*[test/TestServer/TestDeviceConfig.js, line 7 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRi0h2FaA4-5LDZWs2H:-KRi0h2Gl2Wuusglvsmc:b8kko4q) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/64529c3e7c7f6b32f9a806dd80d3d74bebe965de/test/TestServer/TestDeviceConfig.js#L7)):*
> ```JavaScript
>   retryTimeout:    3 * 1000,
>   setupTimeout:    1 * 60 * 1000,
>   testTimeout:     10 * 60 * 1000,
> ```

I think that 2-3 minutes would be enought

---

*[test/TestServer/TestFramework.js, line 21 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRi12paUuOzeo_h-bZs:-KRi12pbgv09tFoIN8qo:b-2dy2qd) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/64529c3e7c7f6b32f9a806dd80d3d74bebe965de/test/TestServer/TestFramework.js#L21)):*
> ```JavaScript
> 
>   this.logger = logger || console;
>   asserts.exists(this.logger);
> ```

This assert makes no sense since `this.logger` has the default value`console`

---

*[test/TestServer/TestFramework.js, line 25 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRi1WdseGDI9Ws6Vp9t:-KRi1Wdtm2KIPplt8siu:bat3tpd) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/64529c3e7c7f6b32f9a806dd80d3d74bebe965de/test/TestServer/TestFramework.js#L25)):*
> ```JavaScript
>   // 'config' provided by the user.
>   // Tells us how many devices we need.
>   asserts.isObject(config);
> ```

Just to clarify the code it would be better to move a `config` validation to a fucntion

---

*[test/TestServer/TestFramework.js, line 44 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRi1yXHsw1BITUb1nMq:-KRi1yXHsw1BITUb1nMr:b8603ea) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/64529c3e7c7f6b32f9a806dd80d3d74bebe965de/test/TestServer/TestFramework.js#L44)):*
> ```JavaScript
>         devices: [],
>         deviceIndexes: {},
>         success: undefined
> ```

 it's strange the success can be something different than `true` or `false`

if there are more than two conditions of success then we probably need an enum

---

*[test/TestServer/TestFramework.js, line 50 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRi2sFmqth5Zosb6kY2:-KRi2sFnwgEQluIMSQIz:bmfpvlb) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/64529c3e7c7f6b32f9a806dd80d3d74bebe965de/test/TestServer/TestFramework.js#L50)):*
> ```JavaScript
>   }, {});
> 
>   this._timer = setTimeout(function () {
> ```

I think that to have `setTimeout` inside `constructor` is not a really good idea
 In this case `TestFramework` looks like a bomb with timer and we should always watch that it won't blow up (I mean wrap it into `try/catch`)
So device connection timeout should be outside the `TestFramework`
Or it can be implemented inside `TestFramework` but like an `event` or something like that

---

*[test/TestServer/TestFramework.js, line 60 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRi3Aww9soNgFlpanXQ:-KRi3Aww9soNgFlpanXR:bpr8zs2) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/64529c3e7c7f6b32f9a806dd80d3d74bebe965de/test/TestServer/TestFramework.js#L60)):*
> ```JavaScript
>   asserts.instanceOf(device, TestDevice);
> 
>   if (!device.supportedHardware) {
> ```

as I know devices that failed native UT must be disqualified too

---

*[test/TestServer/TestFramework.js, line 123 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRi6Dvgw9WmIIsGjOBS:-KRi6Dvhty7OcHTVTaaE:bqq0qh6) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/64529c3e7c7f6b32f9a806dd80d3d74bebe965de/test/TestServer/TestFramework.js#L123)):*
> ```JavaScript
>   });
>   var isCompleted = results.every(function (result) {
>     return result !== undefined;
> ```

The same story as for `success` (see above)

---

*[test/TestServer/UnitTestFramework.js, line 20 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRi6_J8SWT3SYd8J8Du:-KRi6_J8SWT3SYd8J8Dv:b3xyqt7) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/64529c3e7c7f6b32f9a806dd80d3d74bebe965de/test/TestServer/UnitTestFramework.js#L20)):*
> ```JavaScript
>   var self = this;
> 
>   this.logger = logger || console;
> ```

`TestFramework` already has a default value for `logger`, so the `logger` can be simply passed to `UnitTestFramework.super` 

---

*[test/TestServer/UnitTestFramework.js, line 45 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRi7KKB9oAawxIX-X8A:-KRi7KKCI1CIjXjcHWiy:b-6lxqp2) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/64529c3e7c7f6b32f9a806dd80d3d74bebe965de/test/TestServer/UnitTestFramework.js#L45)):*
> ```JavaScript
>   assert(
>     count > 0,
>     'we should have at least one device'
> ```

We should have at least 3 devices
@yaring please correct me if I'm mistaken

---

*[test/TestServer/utils/polyfills.js, line 1 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRi8AXxz755zeECP8nf:-KRi8AXxz755zeECP8ng:bk5a9xe) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/64529c3e7c7f6b32f9a806dd80d3d74bebe965de/test/TestServer/utils/polyfills.js#L1)):*
> ```JavaScript
> 'use strict';
> ```

@aaldiev @yarong Since we have these polyfills why not to use `babel/polyfills` to be able to use ES6 features?

---

*[test/www/jxcore/bv_tests/testThaliMobileNativeDiscoveryCoordinated.js, line 273 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRi8ZpVOQDdwprcGGgz:-KRi8ZpWuX-mAU3Sr24d:bt421oe) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/64529c3e7c7f6b32f9a806dd80d3d74bebe965de/test/www/jxcore/bv_tests/testThaliMobileNativeDiscoveryCoordinated.js#L273)):*
> ```JavaScript
> 
>   setTimeout(function () {
>     Mobile('peerAvailabilityChanged').registerToNative(function () {});
> ```

What for this `registerToNative` ?

---

*[test/www/jxcore/meta_tests/disabled/testUnitTestFramework.js, line 14 at r7](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRi8hInK8DeIFzBbNYr:-KRi8hInK8DeIFzBbNYs:b-io2yng) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/64529c3e7c7f6b32f9a806dd80d3d74bebe965de/test/www/jxcore/meta_tests/disabled/testUnitTestFramework.js#L14)):*
> ```JavaScript
>   },
>   teardown: function (t) {
>     console.log();
> ```

Empty `console.log` :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-:-KRi8nooPb0Cku8Q8Mby:b-ypjnip)*
<!-- Sent from Reviewable.io -->

ThaliTester commented:
PR is added to the queue for testing as 1. task. (f9d1490)

ThaliTester commented:
Test (Fail) 84500654 build is completed (f9d1490)

See https://github.com/ThaliTester/TestResults/tree/84500654f9d1490_New_test_server_andrew-aladev/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (a1aa181)

andrew-aladev commented:
Sorry, not ready with comments yet. I will finish it tomorrow morning.

ThaliTester commented:
Test (Fail) 84500654 build is completed (a1aa181)

See https://github.com/ThaliTester/TestResults/tree/84500654a1aa181_New_test_server_andrew-aladev/ for the logs

andrew-aladev commented:
```
Mobile('peerAvailabilityChanged').registerToNative(function () {});
```
This is a way to unbind current native handler for now.

ThaliTester commented:
PR is added to the queue for testing as 1. task. (d9ce31e)

ThaliTester commented:
Test (Success) 84500654 build is completed (d9ce31e)

See https://github.com/ThaliTester/TestResults/tree/84500654d9ce31e_New_test_server_andrew-aladev/ for the logs

ThaliTester commented:
Test 84500654d9ce31e(d9ce31e) has failed

See https://github.com/ThaliTester/TestResults/tree/84500654d9ce31e_New_test_server_andrew-aladev/ for the fail logs

artemjackson commented:
Reviewed 1 of 7 files at r2, 4 of 10 files at r8, 9 of 14 files at r9, 3 of 4 files at r10.
Review status: 24 of 25 files reviewed at latest revision, 26 unresolved discussions.

---

*[test/TestServer/HttpServer.js, line 73 at r10](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRmimhLX3UDwSsuhppa:-KRmimhMfJ1mKnbBxH6j:b-7xk88h) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/d9ce31eeed472f6fedf1132f7ffaae13288f054f/test/TestServer/HttpServer.js#L73)):*
> ```JavaScript
> 
> Server.prototype._error = function (socket, error) {
>   throw error;
> ```

Shouldn't we log the `error`?
And from context I see that `error` is just a `String`. Shouldn't it be an `Error` instance?

---

*[test/TestServer/Socket.js, line 1 at r10](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRmotIboMfOQ4nlA7j0:-KRmotIciRAqK7r_9B5J:bydxbtj) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/d9ce31eeed472f6fedf1132f7ffaae13288f054f/test/TestServer/Socket.js#L1)):*
> ```JavaScript
> 'use strict';
> ```

Could you also paste a description of the `Socket` life-cycle and may be some examples?

---

*[test/TestServer/Socket.js, line 95 at r10](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRmnNnl6ME2SkuKoP8y:-KRmnNnl6ME2SkuKoP8z:br28mcl) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/d9ce31eeed472f6fedf1132f7ffaae13288f054f/test/TestServer/Socket.js#L95)):*
> ```JavaScript
>   var onceConfirmed;
>   var emitter;
>   data = data || '';
> ```

Empty string `''` is something too simple and it's the same for all data emissions.
I would suggest to use an unique confirmation data for each data emission (e.g. s4, or something like uuid)

---

*[test/TestServer/TestDevice.js, line 32 at r10](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRmpL1Zc409ceYTIwEX:-KRmpL1Zc409ceYTIwEY:b-xpztfb) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/d9ce31eeed472f6fedf1132f7ffaae13288f054f/test/TestServer/TestDevice.js#L32)):*
> ```JavaScript
> }
> 
> TestDevice.prototype._setInfo = function (info) {
> ```

In one place I see `deviceData` but here I see just `info`.
Shouldn't we use a common name (e.g. `deviceInfo`/`deviceData`0 )for these two instances?

---

*[test/TestServer/config/UnitTest.js, line 10 at r10](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRmq927GIWt-7cyFLtr:-KRmq927GIWt-7cyFLts:bb70w76) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/d9ce31eeed472f6fedf1132f7ffaae13288f054f/test/TestServer/config/UnitTest.js#L10)):*
> ```JavaScript
>     android: 3
>   },
>   minDevices: {
> ```

Nice

---

*[test/TestServer/utils/logger.js, line 27 at r10](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRmqYXCf0l2h3RIBach:-KRmqYXCf0l2h3RIBaci:b-lvyq2r) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/d9ce31eeed472f6fedf1132f7ffaae13288f054f/test/TestServer/utils/logger.js#L27)):*
> ```JavaScript
>     now, level.toUpperCase(), meta.tag, message
>   );
>   jxcore.utils.console.log(message);
> ```

Where are we going to use this logger? On jxcore only or not?
If not we should probably use the `console` instead.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007)*
<!-- Sent from Reviewable.io -->

ThaliTester commented:
PR is added to the queue for testing as 2. task. (cf2d3d2)

ThaliTester commented:
Test (Fail) 84500654 build is completed (cf2d3d2)

See https://github.com/ThaliTester/TestResults/tree/84500654cf2d3d2_New_test_server_andrew-aladev/ for the logs

artemjackson commented:
Reviewed 1 of 4 files at r10, 6 of 6 files at r11.
Review status: all files reviewed at latest revision, 30 unresolved discussions.

---

*[test/TestServer/HttpServer.js, line 75 at r11](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRnIUr6bIa0BOqdGiAM:-KRnIUr6bIa0BOqdGiAN:bpwiy8f) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/cf2d3d26153283a243d55a51c1182af133260141/test/TestServer/HttpServer.js#L75)):*
> ```JavaScript
>   logger.error(
>     'uncaught exception, error: \'%s\', stack: \'%s\'',
>     error.toString(), error.stack
> ```

Em... I'm confused
Andry, is error a `String` or an `Object`?
If `String` there is no need to use `.toString()` on it.
If `Object` so we need to specify something reasonable to `new Error()`instead of `error`object

---

*[test/TestServer/Socket.js, line 115 at r11](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRnJfaoUx_Wq61e-Sh3:-KRnJfaoUx_Wq61e-Sh4:bp3cec5) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/cf2d3d26153283a243d55a51c1182af133260141/test/TestServer/Socket.js#L115)):*
> ```JavaScript
>   var emitter;
> 
>   data = uuid.v4() + (data || '');
> ```

I just mean to use `v4()` as default value if no `data` specified :)

---

*[test/TestServer/utils/logger.js, line 27 at r11](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRnKA-0Sr6T2M-GBgKe:-KRnKA-1ubRSGvHspHnu:bvqqky1) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/cf2d3d26153283a243d55a51c1182af133260141/test/TestServer/utils/logger.js#L27)):*
> ```JavaScript
>   ThaliLogger.prototype._logger = jxcore.utils.console.log;
> } else {
>   ThaliLogger.prototype._logger = console.log;
> ```

Should `_logger` be single for all `ThaliLogger` instances and  be in the `prototype` or  should it be unique for each`ThaliLogger` instance and be in the `constructor`?

---

*[test/www/jxcore/lib/thaliTape.js, line 61 at r11](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRnKjvqINotBOrK8TJg:-KRnKjvqINotBOrK8TJh:brvg9x5) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/cf2d3d26153283a243d55a51c1182af133260141/test/www/jxcore/lib/thaliTape.js#L61)):*
> ```JavaScript
> };
> 
> // We should remove prefix (uuid.v4) from data.
> ```

See the comment above about `v4()`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007)*
<!-- Sent from Reviewable.io -->

ThaliTester commented:
PR is added to the queue for testing as 2. task. (ae9737d)

artemjackson commented:
Review status: 23 of 25 files reviewed at latest revision, 34 unresolved discussions.

---

*[test/TestServer/Socket.js, line 115 at r12](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRnQzah-PPgKrE1Bo-6:-KRnQzah-PPgKrE1Bo-7:bx1m52c) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/ae9737d0b828d8ea173225b4cb092150990a30ad/test/TestServer/Socket.js#L115)):*
> ```JavaScript
>   var emitter;
> 
>   data = uuid.v4() + (data || '');
> ```

If it's desarable to send some kind of verification data (e.g. `v4()`) so it should be send separately.
I think we should send something like `verificationValue` and actual `data` separately 

---

*[test/www/jxcore/lib/thaliTape.js, line 64 at r11](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRnPwRkcv8bhY_Ptzu4:-KRnPwRkcv8bhY_Ptzu5:b7xs046) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/cf2d3d26153283a243d55a51c1182af133260141/test/www/jxcore/lib/thaliTape.js#L64)):*
> ```JavaScript
> function parseData(dataString) {
>   assert(
>     dataString.length >= thaliTape.length,
> ```

what is `ThaliTape.length` ???

---

*[test/www/jxcore/lib/thaliTape.js, line 69 at r11](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRnQgugg2b5owwSSirD:-KRnQguhykH9qRRpbr3t:b-cgurf) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/cf2d3d26153283a243d55a51c1182af133260141/test/www/jxcore/lib/thaliTape.js#L69)):*
> ```JavaScript
> 
>   var uuidString = dataString.substring(0, thaliTape.uuid.length);
>   var parsedUuidString = uuid.unparse(uuid.parse(uuidString));
> ```

Emmm... What for this check?

---

*[test/www/jxcore/lib/thaliTape.js, line 95 at r11](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KRnRE0K7qhw5QjoBR71:-KRnRE0K7qhw5QjoBR72:b-tx1q61) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/cf2d3d26153283a243d55a51c1182af133260141/test/www/jxcore/lib/thaliTape.js#L95)):*
> ```JavaScript
>     var success = true;
>     testServer.once('setup_' + name, function (data) {
>       parseData(data);
> ```

Is it just a function call without any  assignment?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007)*
<!-- Sent from Reviewable.io -->

ThaliTester commented:
Test (Success) 84500654 build is completed (ae9737d)

See https://github.com/ThaliTester/TestResults/tree/84500654ae9737d_New_test_server_andrew-aladev/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 3. task. (cb6fb55)

ThaliTester commented:
Test (Fail) 84500654 build is completed (cb6fb55)

See https://github.com/ThaliTester/TestResults/tree/84500654cb6fb55_New_test_server_andrew-aladev/ for the logs

ThaliTester commented:
Test 84500654ae9737d(ae9737d) has failed

See https://github.com/ThaliTester/TestResults/tree/84500654ae9737d_New_test_server_andrew-aladev/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 3. task. (13b5ede)

ThaliTester commented:
Test (Success) 84500654 build is completed (13b5ede)

See https://github.com/ThaliTester/TestResults/tree/8450065413b5ede_New_test_server_andrew-aladev/ for the logs

ThaliTester commented:
Test 8450065413b5ede(13b5ede) has failed

See https://github.com/ThaliTester/TestResults/tree/8450065413b5ede_New_test_server_andrew-aladev/ for the fail logs

artemjackson commented:
Reviewed 1 of 2 files at r12, 1 of 3 files at r13, 2 of 2 files at r14.
Review status: 24 of 26 files reviewed at latest revision, 36 unresolved discussions.

---

*[test/TestServer/Socket.js, line 118 at r14](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KS0K6lxTLKKfeKVG4Mu:-KS0K6lxTLKKfeKVG4Mv:b-lfjc5t) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/13b5ede0278109aa520501b2e823967158b72d35/test/TestServer/Socket.js#L118)):*
> ```JavaScript
>     uuid:    uuid.v4(),
>     content: data || ''
>   });
> ```

Why should we send stringified `Object`? Why not to send a real `Object` instead?

---

*[test/TestServer/utils/logger.js, line 27 at r14](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KS0LAvNltQgxImIh8fF:-KS0LAvNltQgxImIh8fG:b-2gwutc) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/13b5ede0278109aa520501b2e823967158b72d35/test/TestServer/utils/logger.js#L27)):*
> ```JavaScript
>   ThaliLogger._logger = jxcore.utils.console.log;
> } else {
>   ThaliLogger._logger = console.log;
> ```

I mean `this._logger`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007)*
<!-- Sent from Reviewable.io -->

artemjackson commented:
Review status: 24 of 26 files reviewed at latest revision, 36 unresolved discussions.

---

*[test/TestServer/utils/logger.js, line 27 at r14](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-KS0LAvNltQgxImIh8fF:-KS0LhHthRsgwK9R7akk:b-a2ykyf) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/13b5ede0278109aa520501b2e823967158b72d35/test/TestServer/utils/logger.js#L27)):*
<details><summary><i>Previously, artemjackson (Artyom Stepanishchev) wrote…</i></summary>
> I mean `this._logger`
</details>
BTW `_logger` won't work without `console.log.bind(console)`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007)*
<!-- Sent from Reviewable.io -->

ThaliTester commented:
PR is added to the queue for testing as 4. task. (d984f42)

andrew-aladev commented:
Fixed

ThaliTester commented:
Test (Fail) 84500654 build is completed (d984f42)

See https://github.com/ThaliTester/TestResults/tree/84500654d984f42_New_test_server_andrew-aladev/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 2. task. (7e9483f)

ThaliTester commented:
Test (Fail) 84500654 build is completed (7e9483f)

See https://github.com/ThaliTester/TestResults/tree/845006547e9483f_New_test_server_andrew-aladev/ for the logs

artemjackson commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 11 of 27 files reviewed at latest revision, 36 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1007#-:-KS6Sm4Iu4QCa7nRsd_g:bnfp4nl)*
<!-- Sent from Reviewable.io -->

ThaliTester commented:
PR is added to the queue for testing as 7. task. (163f8d4)

ThaliTester commented:
PR is added to the queue for testing as 8. task. (3424b7d)

ThaliTester commented:
Test (Success) 84500654 build is completed (163f8d4)

See https://github.com/ThaliTester/TestResults/tree/84500654163f8d4_New_test_server_andrew-aladev/ for the logs

ThaliTester commented:
Test (Success) 84500654 build is completed (3424b7d)

See https://github.com/ThaliTester/TestResults/tree/845006543424b7d_New_test_server_andrew-aladev/ for the logs

ThaliTester commented:
Test 84500654163f8d4(163f8d4) has failed

See https://github.com/ThaliTester/TestResults/tree/84500654163f8d4_New_test_server_andrew-aladev/ for the fail logs

ThaliTester commented:
Test 845006543424b7d(3424b7d) has failed

See https://github.com/ThaliTester/TestResults/tree/845006543424b7d_New_test_server_andrew-aladev/ for the fail logs
